### ISSUE TITLE
thread 'main' panicked at 'assertion failed: ::glib::types::instance_of::<Widget>(stash.0 as *const _)'

### ISSUE THREAD

**Issue by bram209** [gtk-rs/gtk#397]
Hello,

I needed global key bindings, so I used the C keybinder lib: https://github.com/engla/keybinder
The binding seems to be working fine, but when I try to call is_visible on a window for example, I get an assertion failed for the glib type

```
    keybinder::bind("<Ctrl>A", move || {
        println!("Works!");
        println!("Visible = {}", window.is_visible());
    });
```

output:

```
Works!
thread 'main' panicked at 'assertion failed: ::glib::types::instance_of::<Widget>(stash.0 as *const _)', /home/bram/.cargo/registry/src/github.com-1ecc6299db9ec823/gtk-0.1.0/src/auto/window.rs:28
```

To be honest, I have no idea what could be causing this.
Does anyone have an idea? 

Sorry for my ignorance

**Comment by GuillaumeGomez** [comment]
Can you show us what you do with `window` please? (initialization and all the calls upon it)

**Comment by bram209** [comment]
I used this example to test out the keybinding
https://github.com/gtk-rs/examples/blob/master/src/basic.rs

**Comment by bram209** [comment]
@GuillaumeGomez I created a repo here: https://github.com/bram209/rs-gtk-3.0-keybinder
You can find the gtk initialization at src/example.rs

**Comment by GuillaumeGomez** [comment]
Well, it seems libkeybinder cannot be installed on my mac, which makes it a bit problematic...

**Comment by GuillaumeGomez** [comment]
Replace:

```rust
keybinder::bind("<Ctrl>A", || {
```

with:

```rust
keybinder::bind("<Ctrl>A", move || {
```

**Comment by bram209** [comment]
Yes I know I need to move the closure, i just removed it since i am just calling gtk::main_quit() in the current example.

Do you have any idea why the assertion fails?

**Comment by GuillaumeGomez** [comment]
I think it comes from libkeybinder. The same code with the keybinder replaced by one of the gtk closure works perfectly.
